src/linear_algebra/finsupp.lean

```diff
@@ -395,6 +395,13 @@ lemma total_comap_domain
    (l.support.preimage f hf).sum (λ i, (l (f i)) • (v i)) :=
 by rw finsupp.total_apply; refl
 
+lemma total_on_finset
+  (s : finset α) (f : α → R) (g : α → M) (hf : ∀ a, f a ≠ 0 → a ∈ s):
+  finsupp.total α M R g (finsupp.on_finset s f hf) =
+    finset.sum (finset.filter (λ (a : α), f a ≠ 0) s) (λ (x : α), f x • g x) :=
```

Comment:
Isn't the right hand side the same as `finset.sum s (λ (x : α), f x • g x)`, because in the case `f x = 0`, then `f x • g x = 0` so it doesn't change the sum anyway?

Reply:
Yes, it is. I adapted the lemma accordingly.